Fix exception thrown when removing 1px stroke programmatically

**mapbox-gl-js version**: 0.21.0

Error is also visible at 0.20.0: https://jsfiddle.net/tmcw/yzmz8wL4/

### Steps to Trigger Behavior

 1. When a map's style loads, add a 1px stroke by calling `map.setPaintProperty('building', 'fill-outline-color', '#f00');`
 2. Then immediately remove the stroke by calling `map.setPaintProperty('building', 'fill-outline-color', undefined);`

### Expected Behavior

The stroke is set and removed.

### Actual Behavior

Color parsing throws an uncaught error: `map.js:474Error: Invalid color undefined(…)`

---

Testcase: https://jsfiddle.net/tmcw/nhLd924o/1/

--- a/js/style/style_declaration.js
+++ b/js/style/style_declaration.js
@@ -13,7 +13,7 @@ function StyleDeclaration(reference, value) {
     // immutable representation of value. used for comparison
     this.json = JSON.stringify(this.value);
 
-    var parsedValue = reference.type === 'color' ? parseColor(this.value) : value;
+    var parsedValue = reference.type === 'color' && this.value ? parseColor(this.value) : value;
     this.calculate = MapboxGLFunction[reference.function || 'piecewise-constant'](parsedValue);
     this.isFeatureConstant = this.calculate.isFeatureConstant;
     this.isZoomConstant = this.calculate.isZoomConstant;

--- a/test/js/style/style.test.js
+++ b/test/js/style/style.test.js
@@ -1239,22 +1239,3 @@ test('Style creates correct number of workers', function(t) {
     t.ok(style);
     t.end();
 });
-
-// This test fails due to https://github.com/mapbox/mapbox-gl-js/issues/2886
-// Remove the `{skip: true}` argument to include it once the bug is fixed.
-test('#setPaintProperty', {skip: true}, function (t) {
-    var style = new Style(createStyleJSON({
-        "sources": {
-            "streets": createGeoJSONSource()
-        }
-    }));
-    style.on('load', function () {
-        style.addLayer({ id: 'building', type: 'fill', source: 'streets' });
-        style.setPaintProperty('building', 'fill-outline-color', '#f00');
-        style.update();
-        style.setPaintProperty('building', 'fill-outline-color', undefined);
-        style.update();
-        t.end();
-    });
-});
-

--- a/test/js/style/style_layer.test.js
+++ b/test/js/style/style_layer.test.js
@@ -266,6 +266,26 @@ test('StyleLayer#setPaintProperty', function(t) {
         });
     });
 
+    t.test('can unset fill-outline-color #2886', function (t) {
+        var layer = StyleLayer.create({
+            id: 'building',
+            type: 'fill',
+            source: 'streets',
+            paint: {
+                'fill-color': '#00f'
+            }
+        });
+
+        layer.setPaintProperty('fill-outline-color', '#f00');
+        layer.updatePaintTransitions([], {transition: false}, null, createAnimationLoop());
+        t.deepEqual(layer.getPaintValue('fill-outline-color'), [1, 0, 0, 1]);
+        layer.setPaintProperty('fill-outline-color', undefined);
+        layer.updatePaintTransitions([], {transition: false}, null, createAnimationLoop());
+        t.deepEqual(layer.getPaintValue('fill-outline-color'), [0, 0, 1, 1]);
+
+        t.end();
+    });
+
     t.end();
 });
 

